src/tests/json-stringify.ts

```diff
@@ -0,0 +1,27 @@
+import { doNotExecute, Equal, Expect } from "./utils";
+
+doNotExecute(() => {
+  const result = JSON.stringify(undefined);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
```

Comment:
`result` here should probably be `undefined`!

Reply:
You're right. I hesitated to use a nested conditional type here for this edge case, opting for the simpler typedef (that is still sound), but technically yes we can tighten this up to be undefined if we are 100% sure the arg passed in is undefined (and not just `extends undefined` since any would fit that bill).

Reply:
I think an extra overload makes sense here.

Reply:
Done.